design/authenticators/authn_sut/authn_sut_solution_design.md

```diff
@@ -0,0 +1,283 @@
+Conjur.
+
+## Requirements
+
+The driver behind this effort is to enable integration between Conjur UI and
+PSM. That integration will require a mechanism for generating a single-use token
+that can be used to authenticate to Conjur. The requirements for this effort are
+as follows:
+
+### Feature Spec for Conjur UI-PSM Integration (CNJR-2392)
+
+> Privileged Session Manager (PSM) enables users to log on to remote
+> machines/applications securely through a proxy machine facilitating
+> streamlined and native workflows for the IT admins. Conjur is now going to
+> integrate with PSM providing integrity across Cyberark products and easy
+> access to the Conjur UI via PSM.
+>
+> In order for PSM to connect to the Conjur UI, a new API(s) would be provided
+> that PSM could call to establish the UI session and login the user directly
+> into the main UI dashboard, without showing the initial login screen. More
+> details will be added in the solution design (TBD).
+>
+> The overall UX will remain the same where users will continue to use the
+> existing workflow of logging onto the PVWA, clicking connect on a defined
+> Conjur instance 1 and connecting through PSM straight into the Conjur UI
+> console.
+
+The SUT authenticator will be the mechanism used to authenticate the user to
+Conjur when they click "Connect" on a Conjur instance in PVWA. PSM will generate
+a SUT for the user using their stored username and password or API key. PSM will
+then pass the SUT to the Conjur UI via a URL parameter. The Conjur UI will use
+the SUT to authenticate the user to Conjur and start a session.
+
+## Assumptions and Prerequisites
+
+Assumptions: Defining the assumptions being made in preparing this Solution
+Design. Detailing why these assumptions are valid and/or acceptable.
+Prerequisites: Do we need additional tooling to develop this feature? Do we need
+additional tooling to test this feature in our pipeline? What needs to be in
+place to build this feature?
+
+## Out-of-scope
+
+- SUTs will not be able to be used to authenticate to Conjur via the CLI
+  - SUTs are intended to be used in automated processes. The CLI is intended to
+    be used by humans (developers, operators, etc.)
+- There are many possible use cases for SUTs. This effort will focus on the
+  basic use case necessary for Conjur UI-PSM integration. Additional use cases
+  may be considered in the future.
+- The PSM integration itself will not be part of this effort. This effort will
+  only focus on the SUT authenticator and the API endpoint for generating SUTs.
+  The PSM team will implement the PSM integration in a separate effort in the
+  PSM codebase.
+
+## Design
+
+### High-Level Design
+
+#### Workflows
+
+##### Basic SUT Workflow
+
+The following diagram shows the basic workflow for using a SUT to authenticate
+to Conjur. The first step is to call the
+`GET /authn-sut/<authenticator-name>/<account>/login` endpoint with the user's
```

Comment:
What do you think about having this authenticator avoid the individual `service_id` pattern (as we do with `authn-gcp`). It doesn't seem like there will be any specific details that will necessitate having multiple "versions" of the single-use token authentication workflows.

The pattern we use in the GCP authenticator is to nest the key bits under the authenticator level:

```yml
- !policy
  id: conjur
  body: 
  - !policy:
    id: authn-sut
    body:
    - !webservice

    - !group 
      id: authenticatable
      annotations:
        description: Group with permission to authenticate using this authenticator

    - !permit
      role: !group authenticatable
      privilege: [ read, authenticate ]
      resource: !webservice

    - !webservice
      id: status
      annotations:
        description: Web service for checking authenticator status

    - !group
      id: operators
      annotations:
        description: Group with permission to check the authenticator status

    - !permit
      role: !group operators
      privilege: [ read ]
      resource: !webservice status
```

Reply:
I have no problem with that. I'll update the doc to remove the service-id pattern.